vendor/go.etcd.io/etcd/client/v3/kubernetes.go

```diff
@@ -0,0 +1,176 @@
+package clientv3
+
+import (
+	"context"
+
+	pb "go.etcd.io/etcd/api/v3/etcdserverpb"
+	"go.etcd.io/etcd/api/v3/mvccpb"
+)
+
+func NewKubernetes(c *Client) Kubernetes {
+	return &kubernetes{kv: RetryKVClient(c), watcher: c}
+}
+
+type Kubernetes interface {
+	Get(ctx context.Context, key string, revision int64) (kv *mvccpb.KeyValue, rev int64, err error)
+	List(ctx context.Context, prefix string, revision, limit int64, fromKey string) (kvs []*mvccpb.KeyValue, count, rev int64, err error)
+	Count(ctx context.Context, prefix string) (int64, error)
+	OptimisticCreate(ctx context.Context, key string, value []byte) (success bool, rev int64, err error)
+	OptimisticUpdate(ctx context.Context, key string, value []byte, expectedRevision int64) (kv *mvccpb.KeyValue, success bool, rev int64, err error)
+	OptimisticDelete(ctx context.Context, key string, expectedRevision int64) (kv *mvccpb.KeyValue, success bool, rev int64, err error)
+
+	Watch(ctx context.Context, key string, rev int64, prefix bool) WatchChan
+	RequestProgress(ctx context.Context) error
+
+	// Deprecated: Using leases breaks the implicit Kubernetes-etcd contract.
+	// Lease invalidation deletes multiple keys in one transaction breaking Kubernetes watch assumptions about resourceVersion uniqness.
+	OptimisticCreateWithLease(ctx context.Context, key string, value []byte, lease LeaseID) (success bool, rev int64, err error)
```

Comment:
Will it create a lease if it doesn't exist? Overall, some comments would be useful here and for other methods.

Reply:
I'm considering moving lease manager to etcd. There is no need Kubernetes storage to be aware of leases.